src/main/java/de/tum/in/www1/artemis/domain/Result.java

```diff
@@ -218,14 +219,15 @@ public Result hasFeedback(Boolean hasFeedback) {
     }
 
     /**
-     * 1. set score 2. set successful = true, if score >= 100 or false if not
+     * 1. set score and round it to 4 decimal places
+     * 2. set successful = true, if score >= 100 or false if not
      *
      * @param score new score
      */
     public void setScore(Double score) {
         if (score != null) {
-            this.score = score;
-            this.successful = score >= 100.0;
+            this.score = roundToFourDecimalPlaces(score);
```

Comment:
```suggestion
            this.score = roundToNDecimalPlaces(score, 4);
```

Reply:
`roundToNDecimalPlaces` is actually a private function, that is why I wrote a new function. I can make the function public though.

Reply:
done in 20cd2a8efb4c9630ecd1bfc55392b3ca37e9076c